tracing-attributes/src/lib.rs

```diff
@@ -203,6 +209,38 @@ pub fn instrument(args: TokenStream, item: TokenStream) -> TokenStream {
     .into()
 }
 
+fn skips(args: &AttributeArgs) -> HashSet<Ident> {
+    let mut skips = args.iter().filter_map(|arg| match arg {
+        NestedMeta::Meta(Meta::List(MetaList {
+            ref path,
+            ref nested,
+            ..
+        })) if path.is_ident("skip") => Some(nested),
+        _ => None,
+    });
+    let skip = skips.next();
+
+    // Ensure there's only one skip directive.
+    if let Some(_list) = skips.next() {
+        /*
+        return quote_spanned! {
+            lit.span() => compile_error!("expected only a single `skip` argument!")
+        };
+        */
+        panic!("not sure how to make this return an error -- the examples below return impl ToTokens and get stuck into the output.");
```

Comment:
This should be fixed before merging but I'm not sure what the appropriate way to return an error in a proc macro is.

Reply:
The commented-out code above is correct with regards to how you'd display the error to the user — you want to return a `quote`ed block that expands to the `compile_error!` macro. However, unlike the other functions that return `impl ToTokens`, this function isn't actually returning anything that expands to tokens; it's returning a `HashSet`. I'd probably make this return a `Result<HashSet<Ident>, impl ToTokens>` or something? Then, at the call site, you could match the result and insert the compiler error if it's an `Err`.